Project.toml

```diff
@@ -6,13 +6,17 @@ version = "1.0.2"
 ColorSchemes = "35d6a980-a343-548e-a6ea-1d62b119f2f4"
 Colors = "5ae59095-9a9b-59fe-a467-6f913c188581"
 Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
+ImageFiltering = "6a3955dd-da59-5b1f-98d4-e7296123deb5"
```

Comment:
How heavy is ImageFiltering as a dependency?  We may not want to take that on in a "base" package like PlotUtils (though if it's not too heavy, it may make sense.

Also, this needs a compat bound.

Reply:
All we need from it is direct discrete convolution. If we want to roll our own (there is some MIT code at [FastConv.jl](github.com/aamani/FastConv.jl) and I think in Polynomials.jl).

the equivalent numpy call is `np.convolve(a, b, mode="same")`, so if we can reproduce that ourselves, we can remove the dep.

Reply:
```
julia> @time using ImageFiltering
  4.126747 seconds (6.64 M allocations: 339.804 MiB, 4.60% gc time)
```

IMO, it doesn't make sense to increase load time by 4 seconds...will look into the FastConv code.

Reply:
Here's what I patched in for Polynomials.jl- https://github.com/JuliaMath/Polynomials.jl/blob/master/src/contrib.jl

Reply:
which, we could actually use directly if we're OK with the Polynomials.jl dep

Reply:
```
julia> @time using Polynomials
  0.424075 seconds (667.37 k allocations: 35.860 MiB, 13.88% gc time)
```
which is not too bad.  @mkborregaard would you be OK with adding this?

Reply:
similarly, the code I actually use from Polynomials is just weighted linear fitting. We could potentially write it all ourselves if we want to avoid deps, although it will be mostly copy-pasting code.

Reply:
Honestly I'd rather not add a half second loading time to Plots just to pass a function to clims.

Reply:
Yeah, it's a hard balance to strike.  I'm not overly fond of code reuse, either, but any change to load times here will cascade into major dependent packages, so I just want to be careful.

Reply:
@daschw

Reply:
okay, I'lll work on getting code in-place and dropping the deps. I think at most I would need LinearAlgebra.

Reply:
Stdlibs are fine, since they have no overhead.